KSNM Copycat implementation

dallano opened the pull request:
Co-authored-by: OpheliaXI <heliashelium@gmail.com>

Basis and evidence were gathered from the following capture and wikipedia:
https://youtu.be/-X74Fq3snIc
https://www.dropbox.com/s/4fuzn0b162ctx1u/copycat.rar?dl=0
https://ffxiclopedia.fandom.com/wiki/Copycat

<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?
Changes:
 - Osschaart now uses charm in 30 second intervals moving through the orb holder's party list in order. ID of charm was taken from live capture. No evidence was given that Osschaart also absorbs player's spell list or gains their job trais. More captures potentially needed.
 - Osschaart will use the 2 hour ability of the most recently charmed player's job at around 35% HP as seen in live capture
 - Osschaart will summon the appropriate pet according to the job of the player Osschaart most recently charmed. Ex. Osschaart's wyvern will be summoned if Osschaart charms a dragoon. 
 - Osschaart and his pets were given BCNM mob type
 - Osschaart was given spell list based off of capture. But it is likely this list is incomplete (wiki hints he casts all -aga 3 spells.

## Steps to test these changes
Enter KSNM Copycat inside Waughroon Shrine to test

CriticalXI committed:
Uragnite mixin adjustments

0phie committed:
Merge branch 'staging' into behe_drawin

0phie committed:
Fix Sanity Check

0phie committed:
More Sanities

0phie committed:
Merge pull request #83 from CriticalXI/behe_drawin

Adds region draw ins to ToAU Kings, King Vinegaroon and Behemoth.

TabulaRasaXI committed:
Merge branch 'staging' of https://github.com/AirSkyBoat/AirSkyBoat into AirSkyBoat-staging

soapytoast committed:
Merge pull request #63 from TabulaRasaXI/AirSkyBoat-staging

thisoneworksright

Judaine committed:
Added Maze of Shakhrami

soapytoast committed:
Merge pull request #65 from TabulaRasaXI/beastmen_spawn_time

Added Maze of Shakhrami

Warp-ass committed:
attempt to correct ranged attack crash / bug

RunAwayDSP committed:
Fix Garrison, Add Module for OOE NMs

Fix crystal trade and add garrison feature to the override
removes all the bad nm's that were removed from sql from their respective zone.lua

0phie committed:
Revert "[sql] Longbows to Subskill 1"

0phie committed:
Merge pull request #89 from AirSkyBoat/revert-43-shortbow-subskillids

Revert "[sql] Longbows to Subskill 1"

0phie committed:
Revert "Revert "[sql] Longbows to Subskill 1""

0phie committed:
Merge pull request #90 from AirSkyBoat/revert-89-revert-43-shortbow-subskillids

Revert "Revert "[sql] Longbows to Subskill 1""

soapytoast committed:
Merge pull request #70 from TabulaRasaXI/fix-garrison

Fix Garrison, Add Module for OOE NMs

The-Aerec committed:
Add special cases for Longbows + throwing arrows

The-Aerec committed:
Merge pull request #84 from OpheliaXI/Fix_Outpost_NPC_ID_Shift

Fix Outpost NPC ID Shift for Qufim | Notate Volbow's Overseer

The-Aerec committed:
Merge pull request #91 from The-Aerec/rangeTest

Add special cases for Longbows + throwing arrows

0phie committed:
Fix Magical WS | Fix Magical Mob Move | Fix Magical Breath Move

0phie committed:
Remove prints

Warp-ass committed:
Revert "attempt to correct ranged attack crash / bug"

This reverts commit d3c6701e3b7d0e2d62cc2ec53299c24e62c5eb8d.

Warp-ass committed:
Merge pull request #2 from Dwarcade/buffer

Revert "attempt to correct ranged attack crash / bug"

Warp-ass committed:
Merge pull request #1 from AirSkyBoat/staging

ranged and outpost fixes

0phie committed:
Merge pull request #92 from OpheliaXI/Fix_Mobskills_Final

Fix Magical Mobskills | Fix Breath Skills | Fix Magical Weaponskills

0phie committed:
Fix FTP Calcs for Ability WS Calcs

Warp-ass committed:
Update item_weapon.sql

correct conflict

Warp-ass committed:
Merge pull request #74 from TabulaRasaXI/Scavenge-patch-2

Update item_weapon.sql

Warp-ass committed:
Merge pull request #2 from TabulaRasaXI/staging

merge

Warp-ass committed:
Merge branch 'staging' into staging

Warp-ass committed:
Merge pull request #1 from Dwarcade/staging

pull

Warp-ass committed:
Merge pull request #75 from Scavenge/staging

ranged checks, outposts npc id shift

The-Aerec committed:
Merge pull request #93 from OpheliaXI/Fix_FTP_Abilities

Fix FTP Calcs for Ability WS Calcs

Warp-ass committed:
Merge pull request #76 from AirSkyBoat/staging

fix magical ws, fix mobskills, fix magical mob move, fix magical breath

0phie committed:
Stop Weaponskill that Miss From Initiating Skillchain
Co-authored-by: @Critical

0phie committed:
Limit skillchain magic burst to the appropriate element

0phie committed:
Fix Treasure Pools

0phie committed:
Move Synth Recipies Module Into Testing

The-Aerec committed:
Merge pull request #94 from OpheliaXI/Fix_Skillchain_Sufferino

Fix Skillchain Incorrect Elemental Magic Bursts | Fix Skillchains Forming When A Complete Miss Occurs | Fix Treasure Not Going into Treasure Pools

0phie committed:
Correct TH Formulas Back Down to Expected

0phie committed:
Merge branch 'staging' into fix_some_TH_fomulas

0phie committed:
Fix Linux Startup

0phie committed:
Merge pull request #95 from OpheliaXI/fix_some_TH_fomulas

Correct TH Formulas Back Down to Expected

Frankie-hz committed:
SJ Items & OOE

Co-Authored-By: soapytoast <40971475+soapytoast@users.noreply.github.com>

soapytoast committed:
exp ring era corrections & 75cap era corrections

Co-Authored-By: David Jordan <djorda38@uncc.edu>
Co-Authored-By: Nireya <17558211+nireya@users.noreply.github.com>

0phie committed:
Merge pull request #97 from AirSkyBoat/droplist_update

[sql] Droplist update - SJ Items & Removal of OOE

soapytoast committed:
styling and duration correction.

Co-Authored-By: David Jordan <djorda38@uncc.edu>
Co-Authored-By: Nireya <17558211+nireya@users.noreply.github.com>

soapytoast committed:
Update 75_era_vendors.lua

mistakenly trusted ophelias code

Co-Authored-By: Ophelia <ophelia.somerton@gmail.com>
Co-Authored-By: David Jordan <djorda38@uncc.edu>
Co-Authored-By: Nireya <17558211+nireya@users.noreply.github.com>

The-Aerec committed:
Homepoint fixes in jeuno~

soapytoast committed:
Update 75_era_vendors.lua

more syntax and styling corrections

Co-Authored-By: Ophelia <ophelia.somerton@gmail.com>
Co-Authored-By: David Jordan <djorda38@uncc.edu>
Co-Authored-By: Nireya <17558211+nireya@users.noreply.github.com>

0phie committed:
Merge pull request #99 from The-Aerec/more_npc

Homepoint fixes in jeuno

0phie committed:
Merge pull request #98 from soapytoast/shop-module-corrections

75cap exp ring &  vendor corrections

soapytoast committed:
Merge branch 'staging' of https://github.com/TabulaRasaXI/TabulaRasa into staging

soapytoast committed:
Merge branch 'staging' into content-tag-artisanmoogle

RunAwayDSP committed:
Merge pull request #83 from TabulaRasaXI/content-tag-artisanmoogle

TH, vendor, SC/MB, HP crystal corrections and a little more.

soapytoast committed:
Update npc_list.sql

correct content flag

0phie committed:
Disable Pets Following Across Zonelines

The-Aerec committed:
Merge remote-tracking branch 'LSB_Main/base' into stagingUpdate

The-Aerec committed:
Minor fixes and lua error resolution in module.

soapytoast committed:
Update npc_list.sql

0phie committed:
Build in Wyvern Exclusion

The-Aerec committed:
Merge pull request #102 from OpheliaXI/Fix_Pet_Zoning

Disable Pets Following Across Zonelines

0phie committed:
Merge pull request #103 from The-Aerec/stagingUpdate

LSB Base -> Staging

Aeshur committed:
Update item_basic.sql

0phie committed:
Fix CHR Ratio for Charm Chance

0phie committed:
Fix Magic Accuracy dLvl Mults

The-Aerec committed:
Merge pull request #101 from TabulaRasaXI/contenttag

Update npc_list.sql

0phie committed:
Fix Resist Calculation for Tame

0phie committed:
Cleanup a print statement

0phie committed:
Remove an extra print

0phie committed:
Add a catch for skillType nil for PCs

0phie committed:
Make skillType to nil

0phie committed:
Fix a calculation for Ultra Rare TH Calcs

0phie committed:
Paralysis Should Stop Abilities, Items, and RAs From Execution

The-Aerec committed:
Merge pull request #106 from OpheliaXI/Fix_dLvl_Calcs

Fix Magic Accuracy dLvl Multipliers for MEVA | Added a Catch for Nil Skills to Default to C Rank Accuracy Bonus

0phie committed:
Merge pull request #105 from OpheliaXI/Fix_Charm_CHR_Ratio

Fix CHR Ratio for Charm Chance | Fix Tame Usage

0phie committed:
Merge pull request #104 from The-Aerec/minorFixes

Minor fixes and lua error resolution in module.

0phie committed:
Merge pull request #107 from OpheliaXI/Fix_UR_Calc

Fix a calculation for Ultra Rare TH Calcs

0phie committed:
Merge pull request #108 from OpheliaXI/Paralysis_Affects_Things

Paralysis Should Stop Abilities, Items, and RAs From Execution

CriticalXI committed:
The Sanctuary of Zitah NM Audit

CriticalXI committed:
Ro'Maeve NM Audit

0phie committed:
Add Eva Down to Weight Effect

soapytoast committed:
Merge pull request #1 from AirSkyBoat/staging

from asb

0phie committed:
Fix Restoring onEffect Lose

CriticalXI committed:
Boyahda Tree NM Audit

0phie committed:
Merge pull request #109 from OpheliaXI/Add_EVA_Down_To_Weight

Add Evasion Down to Weight Effect

Judaine committed:
Changed buyback price for guild vendors to 1gil.

soapytoast committed:
Merge pull request #95 from TabulaRasaXI/guild-vendor-hotfix-resale

Changed buyback price for guild vendors to 1gil.

soapytoast committed:
Merge pull request #92 from TabulaRasaXI/rmt-update-prices

Adjust Bird blood and Beastmen blood prices

soapytoast committed:
Air sky boat staging (#96)

* Shetal Stone Latent (No Food) ATT +10 / EVA +10

* Adjust required skills of recipe for Kazaridachi

Bg wiki ffxiclopedia and wiki.ffo.jp all agree though it is likely all were copied from jp wiki

* Disable Abyssea QMs which call deprecated implementation

* Interaction - Mission: Traitor in the Midst


Add default actions


Syntax


Fix bit offset for tracking


Additional bit


Incorrect class reference

* [sql] Add additional NPCs from captures

* Fix Lua Errors When Using Restrict Content

* kuftal digger was accidentally set to aggro in #1795

* Interaction - Mission: Betrayal at Beaucedine

* Interaction - Mission: On Thin Ice

* [cmake] Force download and build LuaJIT on Unix

* [core] [cmake] Adjust FindLuaJIT.cmake to static link libluajit over using the shared object

* [core] [cmake] Attempt to fix libluajit cmake build for OSX

* Update npc_list.sql

correct content flag

* [core] update mob controller to handle Petrification in CheckDetection, facetarget fix

* [core] Adjust SmallPacket0x015 handler to always request updates on all entities

* Adjust logic to add UPDATE_POS mask when rotation changes as well as x/y/z and targid

* Disable Pets Following Across Zonelines

* Update npc_list.sql

* Build in Wyvern Exclusion

* Fix CHR Ratio for Charm Chance

* Fix Magic Accuracy dLvl Mults

* Fix Resist Calculation for Tame

* Cleanup a print statement

* Remove an extra print

* Add a catch for skillType nil for PCs

* Make skillType to nil

* Minor fixes and lua error resolution in module.

* Fix a calculation for Ultra Rare TH Calcs

* Paralysis Should Stop Abilities, Items, and RAs From Execution

* Add Eva Down to Weight Effect

* Fix Restoring onEffect Lose

Co-authored-by: NeuromancerXI <neuromancerxi@users.noreply.github.com>
Co-authored-by: TeoTwawki <arawn.annwn@yandex.com>
Co-authored-by: claywar <claywar@beerlab.dev>
Co-authored-by: claywar <81713309+claywar@users.noreply.github.com>
Co-authored-by: Aerec <ezschneider1@gmail.com>
Co-authored-by: beneliezer <beneliezer@gmail.com>
Co-authored-by: Zach Toogood <zrtoogood@gmail.com>
Co-authored-by: WinterSolstice8 <60417494+WinterSolstice8@users.noreply.github.com>
Co-authored-by: OpheliaXI <heliashelium@gmail.com>
Co-authored-by: Ophelia <84217630+OpheliaXI@users.noreply.github.com>

soapytoast committed:
Update npc_list.sql

set content flag

soapytoast committed:
Update npc_list.sql

correcting to SOA

0phie committed:
Merge branch 'staging' into staging

CriticalXI committed:
Yhoator Jungle and Temple of Uggalepih NM Audit

CriticalXI committed:
Yhoator Jungle and Temple of Uggalepih NM Audit

0phie committed:
Fix Guild Holidays | Remove Era Specific Modules

The-Aerec committed:
Merge pull request #112 from OpheliaXI/fix_guild_shop_menus

Fix Guild Holidays | Remove Non-Era Specific Modules

0phie committed:
Fix the dCHR modifier on charm's success rate

The-Aerec committed:
Merge pull request #113 from OpheliaXI/reformulate_charm

Fix the dCHR modifier on charm's success rate

0phie committed:
Add Crash Persistence to Guild Shop Quantities

0phie committed:
Merge pull request #114 from OpheliaXI/add_persistence_to_guild_shops

Add Crash Persistence to Guild Shop Quantities

soapytoast committed:
Merge pull request #2 from AirSkyBoat/staging

update to base asb

soapytoast committed:
Update npc_list.sql

Set Content tag to Abyssea for symphonic curators. They were not added until 2012

CriticalXI committed:
Den of Rancor and Ifrits Cauldron NM Audit

CriticalXI committed:
Den of Rancor and Ifrits Cauldron NM Audit

CriticalXI committed:
Yuhtunga NM Audit

CriticalXI committed:
Eastern Altepa NM Audit

Judaine committed:
TH Fix for lower TH levels.

Corrected the "bonus" variable type to a float instead of integer, as well as updated calculations for some TH1-4 categories.

CriticalXI committed:
Sea Serpent Grotto NM Audit

CriticalXI committed:
Gustav Tunnel and Cape Terrigan NM Audit

0phie committed:
Add a First Hit Accuracy Bonus to Physical Weaponskills

CriticalXI committed:
Merge pull request #115 from OpheliaXI/Add_First_Hit_Bonus

Add a First Hit Accuracy Bonus to Physical Weaponskills

The-Aerec committed:
Fix module errors

CriticalXI committed:
Kuftal Tunnel NM Audit
Co-Authored-By: Ophelia <heliashelium@gmail.com>

0phie committed:
Add Nil Checks For Resist Rate Calcs | Fix Magical Weaponskill Resist Rate Application

0phie committed:
Fix a Crash Related to Null m_PetType

0phie committed:
Merge pull request #117 from OpheliaXI/Add_First_Hit_Bonus

Add Nil Checks For Resist Rate Calcs | Fix Magical Weaponskill Resist Rate Application

0phie committed:
Merge pull request #111 from soapytoast/npclist-tags

Update npc_list.sql

0phie committed:
Merge pull request #116 from The-Aerec/moreModuleFixes

Fix module errors

CriticalXI committed:
Qufim Island NM Audit

0phie committed:
Merge pull request #118 from OpheliaXI/fix_bst_pet_crash

Fix a Crash Related to Null m_PetType

0phie committed:
Fix Bard Buff Song Power Calculations

0phie committed:
Correct Songs to Era Values or Close Approximations

0phie committed:
Standardize ApplyTH

0phie committed:
Merge pull request #119 from OpheliaXI/fix_bard_buff_power_calcs

Fix Buff Song Power Calcs | Fix Buff Song Power Caps

0phie committed:
Merge pull request #120 from OpheliaXI/Cherry-pick_TH_Mults

Fix TH Mults | Change TH Declaration to Float

CriticalXI committed:
Delkfutt Tower NM audit

CriticalXI committed:
Miseareux Coast NM Audit

CriticalXI committed:
Merge branch 'base' into update_staging

CriticalXI committed:
Merge pull request #125 from CriticalXI/update_staging

LSB Base -> Staging

0phie committed:
Build Extra Checks for Important Pointers and Values

Fix Warnings | Adjust Zone Restriction

Fix Null Check for Zoning

Fix Clang Format Error

Fix Errors

Fix More Clang

More Clang

The-Aerec committed:
Merge pull request #129 from OpheliaXI/Fix_Cluster_Crashes

Build Extra Checks for Important Pointers and Values

0phie committed:
Fix Issues with Astral Flow and Call Wyvern for Mobs Outside of Dynamis

0phie committed:
Change Astral Flow avatarOffsets to global

0phie committed:
Merge pull request #132 from OpheliaXI/Fix_Dyna_Skill_Issue

Fix Issues with Astral Flow and Call Wyvern for Mobs Outside of Dynamis

0phie committed:
Merge pull request #122 from CriticalXI/litelor_audit

Li'Telor Region NM Audit

CriticalXI committed:
Merge pull request #123 from CriticalXI/uplands_audit_asb

Elshimo Uplands Region NM Audit

0phie committed:
Add Additional Nullptr Checks for RA and Superjump | Fix Potential Deref with warp() (#133)

* Add Additional Nullptr Checks for RA and Superjump | Fix Potential Deref with warp()

* Fix Clang Formatting BS

* clang

0phie committed:
Add More Variance to Mob Physical Moves

0phie committed:
Reduce variance to 5% cap

0phie committed:
Merge pull request #126 from CriticalXI/vollbow_audit

Vollbow Region NM Audit

CriticalXI committed:
Merge pull request #134 from OpheliaXI/Add_Variance_To_Mobskills

Add More Variance to Mob Physical Moves

0phie committed:
Audit Enhancing Spells Pt 1

soapytoast committed:
Merge branch 'AirSkyBoat:staging' into staging

soapytoast committed:
Update npc_list.sql

update content flag

CriticalXI committed:
Merge remote-tracking branch 'upstream/base' into staging_update

0phie committed:
Merge pull request #137 from soapytoast/staging

content flag addition

CriticalXI committed:
Correcting clang errors

0phie committed:
Merge pull request #136 from OpheliaXI/Protect_Shell_Audit

Audit Enhancing Spells Pt 1

0phie committed:
Merge pull request #139 from CriticalXI/staging_update

LSB Base -> Staging

0phie committed:
Merge pull request #127 from CriticalXI/lowlands_audit

Elshimo Lowlands NM Audit

The-Aerec committed:
Update Linux CI to include modules

0phie committed:
Merge pull request #142 from The-Aerec/gitCIUpdate

Update Linux CI to include modules

The-Aerec committed:
CI check take two

0phie committed:
Merge pull request #143 from The-Aerec/gitCIUpdate

CI check take two

CriticalXI committed:
Fauregandi Region NM Audit (#128)

* Ranguemont Pass NM Audit

* Beaucedine Glacier NM Audit

* Fei'Yin NM Audit

* Pso'Xja NM Audit

0phie committed:
Add Per Zone Content Flags to Mob Groups | Enable Piping for Specific Group Content Flags (#140)

* Add core piping for mob group content tags

* Add Per Zone Mob Groups Content Flags

* Fix Clang

* More Clang

CriticalXI committed:
Lufaise Meadows NM Audit

CriticalXI committed:
Sacrarium NM Audit

0phie committed:
Rework Avatar Stats, Jobs, Mods, and Skills (#135)

* Change Stat Casing of Avatar Pets | Change Avatar Pet Jobs | Add Fenrir and Diabolos Bonuses

* Fix Clang Formatting Issues

* Audit Shiva

* Audit Garuda

* Audit Titan

* Audit Ramuh

* Audit Leviathan | Audit Carbuncle

* Audit Diabolos

* Fix Leviathan Subjob

* Rework Magical Bloodpacts | Fix a few mods

* Fix Sanity

* Fix more sanity checks

* More Sanity

CriticalXI committed:
Phomuina Aqueducts NM Audit

CriticalXI committed:
Riverne-Site A01 NM Audit

CriticalXI committed:
Riverne Site B01 NM Audit

0phie committed:
Fix Avatar ATK | Add WSC Calc to Physical and Magical Bloodpacts

0phie committed:
Fix Sanity Check

The-Aerec committed:
Merge remote-tracking branch 'LSB_Main/base' into stagingUpdate

CriticalXI committed:
Valdeaunia Region NM Audit (#130)

* Xarcabard NM Audit

* Ulegeurand Range NM Audit

* Castle Zvahl NM Audit

0phie committed:
Merge pull request #146 from The-Aerec/stagingUpdate

Staging update

0phie committed:
Merge pull request #145 from OpheliaXI/More_Avatar_BP_Work

Fix Avatar ATK | Add WSC | Rework Camisado | Fix Inaccuracies in Pacts

0phie committed:
Fix 75_era_spells.sql | Fix WS First Hit Bonus

0phie committed:
Merge pull request #131 from CriticalXI/kuzotz_region

Kuzotz Region NM Audit

0phie committed:
Merge pull request #149 from OpheliaXI/Fix_WS_Fix_MB

Fix 75_era_spells.sql | Fix WS First Hit Bonus

0phie committed:
Fix Damage Ratios for Main, Sub, and Ammo Types (#150)

* Fix Damage Ratios for Main, Sub, and Ammo Types

* Fix Clang

* Fix more clang

0phie committed:
Add Spell Cast Time Modifier Using Local Var

0phie committed:
Make Var More Specific

0phie committed:
Remove Testing Stuff

0phie committed:
Merge pull request #151 from OpheliaXI/Add_Spell_Cast_Time_Mod

Add Spell Cast Time Modifier Using Local Var

0phie committed:
Merge pull request #141 from CriticalXI/qufim_audit

Qufim Island Region NM Audit

0phie committed:
Add MAGIC_INTERRUPTED State Listener (#153)

0phie committed:
Add !clearsession  | Add !spin (#152)

* Add clearSession

Co-authored-by: @Aerec

* Update setRotation to Sent Char Update Packet

* Add command luas for spin and clearsession

* Update spin help message

* Fix Sanity

* Fix Errors

* Add Newline to Spin

The-Aerec committed:
Fix Linux Error

0phie committed:
Merge pull request #155 from The-Aerec/errorFix

Fix Linux Error

0phie committed:
Fix Rage Mixin | Fix Claim Shield | Add Selective Lottery Persistence (#154)

* Fix Rage Mixin

* Add Rage 100% HP Removal

* Fix sanity check

* Convert Claim Shield to Mixin

Co-authored-by: zach2good <zrtoogood@gmail.com>

* Add Lottery NM Persistence Module | Update Claim Shield to Have Shields

Co-authored-by: zach2good <zrtoogood@gmail.com>

CriticalXI committed:
Merge pull request #144 from CriticalXI/tavnazia_audit

Tavnazia Archipelago Region NM Audit

0phie committed:
Add Pet Dmg Multipliers For Targets With a Mod (#156)

* Add Pet Dmg Multipliers For Targets With a Mod

* Add Physical Damage Mult Mod to Avatar Physical Bloodpacts

* Fix Clang

* Fix Typo

* Change calculations for the mod

* Further adjustments

soapytoast committed:
Merge pull request #3 from AirSkyBoat/staging

update staging

soapytoast committed:
Update gardening_results_era.sql

deleting the result all together likely causes issues, setting weight to 0 instead.

soapytoast committed:
Update gardening_results_era.sql

soapytoast committed:
Update gardening_results_era.sql

Warp-ass committed:
fix rycharde the chef

returns missing from event triggers

0phie committed:
Merge pull request #158 from soapytoast/staging

change gardening results to 0 weight instead of removal

CriticalXI committed:
Add content tags to NMs and add ENABLE_SYNERGY for Synergy mobs/NPCs (#157)

0phie committed:
Synth Utils Standardization to Era | Add Break Types

Co-authored-by: Frankie-hz <105882754+Frankie-hz@users.noreply.github.com>
Co-authored-by: Simseru <mtanz89@yahoo.com>

0phie committed:
Fix Issues With Status Weaponskills

dallano committed:
KSNM Copycat implementation

Co-authored-by: OpheliaXI <heliashelium@gmail.com>

Basis and evidence were gathered from the following capture and wikipedia:
https://youtu.be/-X74Fq3snIc
https://www.dropbox.com/s/4fuzn0b162ctx1u/copycat.rar?dl=0
https://ffxiclopedia.fandom.com/wiki/Copycat

Changes:
 - Osschaart now uses charm in 30 second intervals moving through the orb holder's party list in order. ID of charm was taken from live capture.
 - Osschaart will use the 2 hour ability of the most recently charmed player's job at around 35% HP as seen in live capture
 - Osschaart will summon the appropriate pet according to the job of the player Osschaart most recently charmed. Ex. Osschaart's wyvern will be summoned if Osschaart charms a dragoon. No evidence was given that Osschaart also absorbs player's spell list or gains their job trais. More captures needed.
 - Osschaart and his pets were given BCNM mob type
 - Osschaart was given spell list based off of capture. But it is likely this list is incomplete

dallano committed:
Style fixes

Cleaned up code:
removed comments
fixed variable declartion
removed unused function

dallano committed:
Update Osschaart.lua

Used enums for 2 hour ability table

dallano committed:
Update Osschaart.lua

Added null eagle eye shot to prevent reading of incorrect jobs.

0phie committed:
Merge pull request #162 from OpheliaXI/fix_status_weaponskills

Fix Issues With Status Weaponskills

0phie committed:
Merge pull request #161 from OpheliaXI/synth_utils_work

Synth Utils Standardization to Era | Add Break Types

0phie committed:
Merge pull request #160 from Scavenge/patch-1

fix rycharde the chef

dallano committed:
Update Osschaart.lua

removed updateEnmity as it's not needed.

CriticalXI committed:
Adds KS99 Early Bird Catches the Wyrm
Co-Authored-By: Lau <neil_mitchell420@hotmail.com>
Co-Authored-By: Namidea <eboh_vandroiy@hotmail.com>
Co-Authored-By: mgramolini <mattygs@gmail.com>

0phie committed:
Fix An Issue With dCHR Calcs in  GetCharmChance()

0phie committed:
Merge pull request #165 from OpheliaXI/fix_charm_dstat_0

Fix An Issue With dCHR Calcs in  GetCharmChance()

0phie committed:
Fix an issue where charm % could be negative (#166)

claywar commented on scripts/globals/items.lua:
👍

claywar commented on sql/mob_skills.sql:
Is this the only place this mobskill is used?

TeoTwawki commented on scripts/globals/mobskills/charm_copycat.lua:
`mob:getID()` can be cast to a variable so that you call it only once, reducing the number of times the code jumps from lua to core and back.

TeoTwawki commented on scripts/zones/Waughroon_Shrine/npcs/Armoury_Crate.lua:
our pattern is to use `_P1` for ` +1` and `_M1` for ` -1`

dallano commented on sql/mob_skills.sql:
Yes, as far as I could tell it was unique to Osschaart. Was also the ID being used in the capture

dallano committed:
General fixes

Changes:
- Added proper naming of SPOOL_OF_GOLD_THREAD to other entities that were previously using only GOLD_THREAD
- Updated name of VILE_ELIXER_1 to _P1
- updated mob:getID() to fetch a local variable

0phie committed:
Merge pull request #164 from CriticalXI/early_bird_catches_the_wyrm

Adds KS99 Early Bird Catches the Wyrm

0phie committed:
Remove additional stat casing for non-normal mobs

0phie committed:
Remove unused vars | Fix Clang

CriticalXI committed:
Implement KS99 The Hills Are Alive

CriticalXI committed:
Merge pull request #167 from OpheliaXI/mob_stat_casing

Remove additional stat casing for non-normal mobs

0phie committed:
Fix Charm for BST Main | Add Piping for CHARMRES (#168)

* Fix Charm for BST Main | Add Piping for CHARMRES

* Fix Clang

0phie committed:
Rework Current Digging System | Pull Xaver-Red Work + LB Chocobo Digging (#163)

* Cherry-Pick Xaver-Red's WIP Chocobo Digging Based on LB

* Core: Send animation to characters in range

* Remove Multiple Dig Code | Make Players Able to Dig in Zone After Zone Pool Is Gone

* Fix Sanity Checks

* Fix issues

Co-authored-by: Xaver-DaRed <xaver.entropia@gmail.com>

CriticalXI committed:
Implement KS99 Horns of War
Co-Authored-By: Zelita <siknoz@yahoo.com>

CriticalXI committed:
Add content check to NMs checked in zone files

0phie committed:
Add Elemental SDT (EEM) Piping | Finish Magic Hit Rate

0phie committed:
Merge pull request #171 from CriticalXI/error_fix

Startup error fixes

0phie committed:
Merge pull request #169 from CriticalXI/the_hills_are_alive

Implement KS99 The Hills Are Alive

0phie committed:
Merge pull request #170 from CriticalXI/horns_of_war

Implement KS99 Horns of War

0phie committed:
Update and fix conflicts

0phie committed:
Merge pull request #172 from OpheliaXI/magic_accuracy_add_SDT

Add Elemental SDT (EEM) Piping | Finish Magic Hit Rate

CriticalXI committed:
Implement KS30 Today's Horoscope
Co-Authored-By: siknoz <siknoz@yahoo.com>

CriticalXI committed:
Shell We Dance adjustments

CriticalXI committed:
Implement ENM Pulling the Plug

CriticalXI committed:
Merge pull request #175 from CriticalXI/uragnite_mixin_asb

Uragnite mixin adjustments

0phie committed:
Fix Magic Burst Rate Calculation

CriticalXI committed:
Merge pull request #179 from OpheliaXI/fix_magic_casts

Fix Magic Burst Rate Calculation

0phie committed:
Fix UDMGMAGIC Mod Application in damage_spell.lua

CriticalXI committed:
Merge pull request #180 from OpheliaXI/fix_magic_casts

Fix UDMGMAGIC Mod Application in damage_spell.lua

CriticalXI committed:
Implement Quest Uninvited Quests
Co-Authored-By: Tiberon <<JosephKeyes75@gmail.com>

0phie committed:
Add Avatar Enmity From BPs

0phie committed:
Merge pull request #182 from OpheliaXI/Fix_Avatar_Hate

Add Avatar Enmity From BPs

0phie committed:
Merge pull request #177 from CriticalXI/pulling_the_plug

Implement ENM Pulling the Plug

0phie committed:
Merge pull request #176 from CriticalXI/shell_we_dance

Shell We Dance ENM Adjustments

0phie committed:
Merge pull request #174 from CriticalXI/todays_horoscope

Implement KS30 Today's Horoscope

0phie committed:
Merge pull request #181 from CriticalXI/uninvited_guests

Implement Quest Uninvited Quests

CriticalXI committed:
Adjustments to Sheep in Antlions Clothing
Co-Authored-By: mgramolini <mattygs@gmail.com>
Co-Authored-By: Caelic <bclynn96@gmail.com>

CriticalXI committed:
Adjustments to ENM Brothers

0phie committed:
Change !pos to include rot in print (#185)

Frankie-hz committed:
mp_drainkiss mp fix

CriticalXI committed:
Merge pull request #183 from CriticalXI/sheep_in_antlions_clothing

Sheep in Antlions Clothing Adjustments

CriticalXI committed:
Merge pull request #184 from CriticalXI/brothers

Adjustments to ENM Brothers

CriticalXI committed:
Merge pull request #186 from AirSkyBoat/mp_drainkiss

MP Drainkiss MP Fix

CriticalXI committed:
Merge pull request #187 from Frankie-hz/Frank-CoP_3-3_The_Road_forks

[Mission] CoP3-3 The Road Forks - Balance Fixes

CriticalXI committed:
Merge remote-tracking branch 'upstream/base' into staging_update

0phie committed:
Merge pull request #209 from CriticalXI/staging_update

LSB Base -> ASB

dallano committed:
Updates

Fixes:
Gave the following new skill list IDs as they were being used twice as well as adjusted their skill list being used in mob_pools
- Osschart
- Snoll Tzar
- Shikaree X, Y, Z

- Removed repeated values in xi.items table
- Fixed bad rebase values (returned charm to use anim ID 437 and 0 casting time
- Removed white spacing in mob_skill_lists as it wasn't following syntax from the rest of the table

Changes:
Added a table lookup for Osschaart's charm ability

dallano committed:
File removal

Removed unnecessary lua mob files

dallano committed:
ESS

Changes:
- Added eagle eye shot to Osschaart's potential 2hr abilities
